api/src/main/java/com/velocitypowered/api/proxy/server/RegisteredServerState.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2018 Velocity Contributors
+ *
+ * The Velocity API is licensed under the terms of the MIT License. For more details,
+ * reference the LICENSE file in the api top-level directory.
+ */
+
+package com.velocitypowered.api.proxy.server;
+
+import com.velocitypowered.api.proxy.Player;
+import com.velocitypowered.api.proxy.messages.ChannelMessageSink;
+import java.util.Collection;
+import net.kyori.adventure.audience.Audience;
+
+/**
+ * A server's current state. This differs from {@link RegisteredServer} for providing a stateful view
+ * on a backend server instance.
+ *
+ * <p>The {@code Audience} associated with a {@code RegisteredServerState} represent all players on the server
+ * connected to this proxy and do not interact with the server in any way.
+ * </p>
+ */
+public interface RegisteredServerState extends ChannelMessageSink, Audience {
```

Comment:
Perhaps a tad controversial, but I would call this `ServerState`. This decouples the notion of registration from the actual purpose of the class: contain the state of a server. A comment below (or above, don't know how GH will order them :P) hints at the fact that we are conflating two different concepts of "registration": one, the server's parameters like e.g. IP:port pair being known by Velocity, and these parameters appearing in `velocity.toml`. I wish we could go further and rename `RegisteredServer` (since Velocity should be the only one creating objects of this type, a `RegisteredServer`/`Server` is known to it by nature), but that's not an option due to API compat.